src/spdx_tools/spdx/validation/spdx_id_validators.py

```diff
@@ -44,7 +48,7 @@ def is_external_doc_ref_present_in_document(external_ref_id: str, document: Docu
 
 
 def validate_spdx_id(
-    spdx_id: str, document: Document, check_document: bool = False, check_files: bool = False
+        spdx_id: str, document: Document, all_spdx_ids: Set[str], check_document: bool = False, check_files: bool = False
```

Comment:
Should you not also have an files_spdx_ids: Set[str] argument for the check_files case? My SPDX file has over 70K entries.   The check_document seems redundant here, since you could simply make all_spdx_ids: Optional[Set[str]] instead and go by whether it was passed as None or not.